notify/jira/jira.go

```diff
@@ -387,3 +395,67 @@ func (n *Notifier) doAPIRequestFullPath(ctx context.Context, method, path string
 
 	return responseBody, false, nil
 }
+
+// deepCopyWithTemplate returns a deep copy of a map/slice/array/string/int/bool or combination thereof, executing the
+// provided template (with the provided data) on all string keys or values. All maps are connverted to
+// map[string]interface{}, with all non-string keys discarded.
+// deepCopyWithTemplateFunc returns a deep copy of a map/slice/array/string/int/bool or combination thereof,
+// executing tmplTextFunc on all string keys or values. All maps are converted to map[string]interface{},
+// with all non-string keys discarded.
+func deepCopyWithTemplate(value interface{}, tmplTextFunc templateFunc) (interface{}, error) {
```

Comment:
This seems generic enough to be moved to a shared package so more integrations can benefit from it.
For example it can be used for PagerDuty in #3944